zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter.launcher;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.interpreter.remote.RemoteInterpreterUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Spark specific launcher.
+ */
+public class SparkInterpreterLauncher extends ShellScriptLauncher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SparkInterpreterLauncher.class);
+
+  public SparkInterpreterLauncher(ZeppelinConfiguration zConf) {
+    super(zConf);
+  }
+
+  @Override
+  protected Map<String, String> buildEnvFromProperties() {
+    Map<String, String> env = new HashMap<String, String>();
+    Properties sparkProperties = new Properties();
+    String sparkMaster = getSparkMaster(properties);
+    for (String key : properties.stringPropertyNames()) {
+      if (RemoteInterpreterUtils.isEnvString(key)) {
+        env.put(key, properties.getProperty(key));
+      }
+      if (isSparkConf(key, properties.getProperty(key))) {
+        sparkProperties.setProperty(key, toShellFormat(properties.getProperty(key)));
+      }
+    }
+
+    setupPropertiesForPySpark(sparkProperties);
+    setupPropertiesForSparkR(sparkProperties, properties.getProperty("SPARK_HOME"));
+    if (isYarnMode() && getDeployMode().equals("cluster")) {
+      env.put("SPARK_YARN_CLUSTER", "true");
```

Comment:
thinking more about this, I think we should avoid Zeppelin specific env with a `SPARK_` prefix, to avoid potential conflict